hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -89,7 +84,15 @@ public String getUuid() {
 
     @Override
     public boolean isAlive() {
```

Comment:
The client does not use connection.write but it always sends the message to the client using AbstractMessageTask.sendClientMessage which does this: final Connection endpointConnection = findSendConnection(); 

We need this kind of logic since for the listeners the connection to the client may change during the client lifetime when the client reconnects to the cluster before it is cleaned up and findSendConnection ensures to use the correct connection.